src/libs.rs

```
@@ -0,0 +1,104 @@
+use std::env;
+use std::error::Error;
+use std::fs;
+
+pub struct Config {
+    pub query: String,
+    pub file_path: String,
+    pub not_case_sensitive: bool,
+}
+
+impl Config {
+    pub fn build(args: &Vec<String>) -> Result<Config, &'static str> {
+        if args.len() < 3 {
+            return Err("Not enough arguments");
+        }
+
+        let query = args[1].clone();
+        let file_path = args[2].clone();
+        let not_case_sensitive =
+            env::var("NOT_CASE_SENSITIVE").is_ok() || args.contains(&String::from("-cs"));
+
+        Ok(Config {
+            query,
+            file_path,
+            not_case_sensitive,
+        })
+    }
+}
+
+pub fn read_from_file(file_path: &String) -> Result<String, Box<dyn Error>> {
+    let contents = fs::read_to_string(file_path).expect("Something went wrong reading the file");
```

What happens if the file is 100GB?

---

Can we return the error instead of `expect`-ing? After all the function returns a `Result<_,Box<dyn Error>>`.